source/detail/parser/x86_64.cpp

```diff
@@ -128,6 +129,29 @@ namespace smeagle::x86_64 {
     return result.str();
   }
 
+  // Get directionality from argument type
+  std::string getDirectionalityFromType(st::Type *paramType) {
+    auto dataType = paramType->getDataClass();
+    std::string dataTypeStr = st::dataClass2Str(dataType);
+
+    // If it's a pointer, we need to know what it's pointing to!
+    if (dataTypeStr == "pointer" || dataTypeStr == "reference") {
+      auto pointerType = paramType->getPointerType();
+
+      // if typo is pointer but type is primitive: imported
+      std::regex is_primitive("(int|char|bool|float|double|null|none|undefined)");
```

Comment:
We can use Dyninst to do this for us. Refactoring it into its own function would be good since we'll need it in several spots, as well.

Reply:
I tried doing this locally but it was the wrong type, so I'll need your help here.